cmd/snap-confine/mount-support.c

```diff
@@ -310,20 +310,55 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		};
 		for (const char **dirs = dirs_from_core; *dirs != NULL; dirs++) {
 			const char *dir = *dirs;
-			struct stat buf;
-			if (access(dir, F_OK) == 0) {
-				sc_must_snprintf(src, sizeof src, "%s%s",
-						 config->rootfs_dir, dir);
-				sc_must_snprintf(dst, sizeof dst, "%s%s",
-						 scratch_dir, dir);
-				if (lstat(src, &buf) == 0
-				    && lstat(dst, &buf) == 0) {
-					sc_do_mount(src, dst, NULL, MS_BIND,
-						    NULL);
-					sc_do_mount("none", dst, NULL, MS_SLAVE,
-						    NULL);
+			if (access(dir, F_OK) != 0) {
+				continue;
+			}
+			struct stat dst_stat;
+			struct stat src_stat;
+			sc_must_snprintf(src, sizeof src, "%s%s",
+					 config->rootfs_dir, dir);
+			sc_must_snprintf(dst, sizeof dst, "%s%s",
+					 scratch_dir, dir);
+			if (lstat(src, &src_stat) != 0) {
+				if (errno == ENOENT) {
+					continue;
+				}
+				die("cannot stat %s from desired rootfs", src);
+			} else {
```

Comment:
You don't need the else part, do you? It just increases indent without any semantic change since you either continue in the loop or die in the if-part above.